File: backend/src/main/java/fr/jbellion/shoppinglist/service/CategoryService.java
package fr.jbellion.shoppinglist.service;

import fr.jbellion.shoppinglist.dto.CategoryDto;

import java.util.List;

public interface CategoryService {

    List<CategoryDto> getAllCategories();

    CategoryDto createCategory(CategoryDto category);

    CategoryDto updateCategory(CategoryDto category);

    void deleteCategory(long id);
}

File: frontend/src/app/services/category.service.ts
import {inject, Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {CategoryModel, CreateCategoryModel} from "../models/category.model";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class CategoryService {

  private readonly http = inject(HttpClient);

  getAllCategories(): Observable<CategoryModel[]> {
    return this.http.get<CategoryModel[]>('/categories');
  }

  createCategory(category: CreateCategoryModel): Observable<CategoryModel> {
    return this.http.post<CategoryModel>('/categories', category);
  }

  updateCategory(category: CategoryModel): Observable<CategoryModel> {
    return this.http.put<CategoryModel>(`/categories/${category.id}`, category);
  }

  deleteCategory(id: number): Observable<void> {
    return this.http.delete<void>(`/categories/${id}`);
  }
}
